atsign-foundation/at_libraries
# feat: apkam-aware activate_cli

**- What I did**
- feat: 'activate' CLI is now APKAM-aware, and supports 
  - onboarding (as before)
  - submitting enrollment requests
  - listing / approving / denying / revoking enrollment requests
  - generating one-time passcodes
  - setting semi-permanent passcode

**- How I did it**
- New code in src/cli : auth_cli.dart, auth_cli_args.dart
- Modified existing bin/activate_cli.dart so it uses auth_cli.main instead of activate_cli.main

**- How to verify it**
- See NoPorts e2e tests ([initial PR](https://github.com/atsign-foundation/noports/pull/1037)) which use this CLI extensively
- Some examples if you wish to test manually:
  - `dart bin/activate_cli.dart -h` to see overall usage info and list of available commands
  - `dart bin/activate_cli.dart <some command> -h` to see help and usage info for a specific command
  - `dart bin/activate_cli.dart otp -a @alice` to generate an OTP for example
  - `dart bin/activate_cli.dart interactive -a @alice` to run interactively

**-Additional context**
- TODO but will be in follow-up PR as it's not on critical path right now: Add code which allows user to 'resume' an enrollment similar to what has been done in at_client_mobile - basically if the CLI is killed while waiting for its enrollment request to be approved or denied, then allow the exact same command to be re-run, but instead of submitting a new enrollment request it finds the details of the previous one and then goes straight to attempting PKAM auth
